test/integration/kubernetes_deploy_test.rb

```diff
@@ -320,7 +320,7 @@ def test_dynamic_erb_collection_works
   # https://github.com/kubernetes/kubernetes/issues/42057 shows how this manifested for a particular field,
   # and although that particular case has been fixed, other invalid specs still aren't caught until apply.
   def test_multiple_invalid_k8s_specs_fails_on_apply_and_prints_template
-    result = deploy_fixtures("hello-cloud", subset: ["web.yml.erb"]) do |fixtures|
+    result = deploy_fixtures("hello-cloud", subset: ["web.yml.erb", "secret.yml"]) do |fixtures|
```

Comment:
I gave you bad advice about which test to use. Turns outs this test only checks the case where all of the errors match in `find_bad_files_from_kubectl_output` so there is functionally no change here. 

It looks like the test that exercises `(filenames_with_sensitive_content - server_dry_run_validated_resource).present?` is `test_apply_failure_with_sensitive_resources_hides_raw_output` except the only resource it has in a secret. So I think we need a new test. Something like: 


```
  def test_apply_failure_with_sensitive_resources_does_not_hide_apply_error
    result = deploy_fixtures("hello-cloud", subset: %w(redis.yml secret.yml))
    assert_deploy_success(result)

    result = deploy_fixtures("hello-cloud", subset: %w(redis.yml secret.yml)) do |fixtures|
      service = fixtures["redis.yml"]["Service"].detect { |s| s.dig("metadata", "name") == "redis-endpoints" }
      service['spec']['clusterIP'] = ''
    end

    assert_deploy_failure(result)
    assert_logs_match('The Service "redis-endpoints" is invalid: spec.clusterIP: Invalid value: "": field is immutable')
    refute_logs_match(/WARNING: .* The raw output may be sensitive and so cannot be displayed./)
    refute_logs_match("password")
  end
```

the redis.yml might not be the best fixture, I was just quickly looking for something with an immutable field. (e.g I'm testing the apply is failing not the template validation).